cpp/CMakeLists.txt

```diff
@@ -147,6 +147,8 @@ add_library (bond
     ${precompiled_sources}
     ${headers})
 
+target_compile_definitions (bond PUBLIC -DBOND_LIB_TYPE=BOND_LIB_TYPE_STATIC)
```

Comment:
We should be consistent with this across our compilation units, so I think we want to set this in Config.cmake like we do with BOOST_ALL_NO_LIB.

Reply:
Wait. It's `PUBLIC`, so it will be propagated automatically. This is fine.